### PR TITLE
Fix cbmc crash on pointer checks of void pointer dereferences

### PR DESCRIPTION
**Pull request by danpoe**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: regression/cbmc/void_pointer6/test.desc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CORE
+main.c
+--pointer-check
+^EXIT=10$
```
Comment (peterschrammel) [inline regression/cbmc/void_pointer6/test.desc:4]: + verification failed?

File: src/analyses/goto_check.cpp [hunk 2]
```diff
@@ -1164,10 +1164,20 @@ void goto_checkt::pointer_validity_check(
...
   const exprt &pointer=expr.pointer();
 
-  auto size_of_expr_opt = size_of_expr(expr.type(), ns);
-  CHECK_RETURN(size_of_expr_opt.has_value());
+  exprt size;
 
-  auto conditions = address_check(pointer, size_of_expr_opt.value());
+  if(expr.type().id() == ID_empty)
+  {
+    size = from_integer(1, size_type());
```
Comment (hannes-steffenhagen-diffblue) [inline src/analyses/goto_check.cpp:1175]: Should we change `size_of_expr` to return `1` here instead?

Comment (danpoe) [reply src/analyses/goto_check.cpp:1175]: Not sure. There are quite a number of uses in the codebase that check that the return value of `size_of_expr()` is not empty, and if it is throw an exception or have some other handling. Probably we should review those at some point, though that's out of scope of this PR.

Comment (peterschrammel) [reply src/analyses/goto_check.cpp:1175]: Is the purpose of this to create an address check that fails?

Comment (danpoe) [reply src/analyses/goto_check.cpp:1175]: No the address check should succeed. Basically `*p` with `p` being a pointer to void is valid as long as `p` points to valid memory, though then using the value would be undefined behavior.

File: regression/cbmc/void_pointer6/test.desc (hunk 1 above)
Comment (danpoe) [reply regression/cbmc/void_pointer6/test.desc:4]: Not necessary I think as `EXIT=10` already indicates that cbmc exited without error and verification failed.

File: src/analyses/goto_check.cpp (hunk 2 above)
Comment (peterschrammel) [reply src/analyses/goto_check.cpp:1175]: Please explain the magic `1` in a comment.

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**